school_center.py

```diff
@@ -39,34 +39,22 @@ def centers_within_distance(school: Dict[str, str], centers: Dict[str, str], dis
     def center_to_dict(c, distance):
         return {'cscode': c['cscode'], 'name': c['name'], 'address': c['address'], 'capacity': c['capacity'], 'lat': c['lat'], 'long': c['long'], 'distance_km': distance}
     
-    def sort_key(c):
-        # intent: sort by preference score DESC then by distance_km ASC 
-        # leaky abstraction - sorted requires a single numberic value for each element
-        return c['distance_km'] * random.uniform(1,5) - get_pref(school['scode'], c['cscode'])*100
```

Comment:
removing this will remove randomization and sort by preference. see comment on what this was trying to acheive